registry/storage/driver/filesystem/driver.go

```diff
@@ -224,6 +224,41 @@ func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) e
 	}
 
 	err := os.Rename(source, dest)
+	// os.Rename will not work across devices, in that case, try to copy then delete instead
+	if err != nil {
+		err = d.copyFile(source, dest)
```

Comment:
Also, this error should be checked against `os.IsExist`, which should be ignored.